compiler/src/dotty/tools/dotc/reporting/diagnostic/messages.scala

```diff
@@ -1968,16 +1968,18 @@ object messages {
     }
   }
 
-  case class BadSymbolicReference(
-    location: String,
-    name: String,
-    denotationOwner: Symbol,
-    src: String
-  )(implicit ctx: Context) extends Message(BadSymbolicReferenceID) {
+  case class BadSymbolicReference(denot: SymDenotation)(implicit ctx: Context) extends Message(BadSymbolicReferenceID) {
     val kind = "Reference"
+
+    private val file = denot.symbol.associatedFile
+    private val (location, src) =
+      if (file != null) (s" in $file", file.toString)
+      else ("", "the signature")
+    private val name = ctx.fresh.setSetting(ctx.settings.YdebugNames, true).nameString(denot.name)
+
     val msg =
       hl"""Bad symbolic reference. A signature$location
-          |refers to $name in ${denotationOwner.showKind} ${denotationOwner.showFullName} which is not available.
+          |refers to $name in ${denot.owner.showKind} ${denot.owner.showFullName} which is not available.
```

Comment:
Here you can also store the result of `owner` in a temporary `val` within the block